functions/src/process-email-queue.ts

```diff
@@ -0,0 +1,79 @@
+import * as functions from "firebase-functions";
+import { firestore } from "firebase-admin";
+import * as mailgun from "mailgun-js";
+import { Config, isEnabled } from "./config";
+import { isAlreadyExistsError } from "./already-exists";
+import { DocumentSnapshot } from "firebase-functions/lib/providers/firestore";
+
+const config = functions.config() as Config;
+const mailgunClient = mailgun({
+  domain: config.mailgun.domain,
+  apiKey: config.mailgun.api_key,
+  host: config.mailgun.host
+});
+const db = firestore();
+
+export const EMAILS_TO_SEND_COLLECTION_NAME = "emails-to-send";
+export const EMAILS_SENT_COLLECTION_NAME = "emails-sent";
+
+export type QueuedEmail = {
+  message: mailgun.messages.SendData;
+};
+
+export const enqueue = async (message: mailgun.messages.SendData) => {
+  await db.collection(EMAILS_TO_SEND_COLLECTION_NAME).add({ message });
+};
+
+const processEmail = async (emailSnapshot: DocumentSnapshot) => {
+  if (!isEnabled(config.features.emails)) {
+    console.info("feature is disabled; aborting");
+    return;
+  }
```

Comment:
I feel it hard to read without space between blocs, I have difficulties to distinguish different bloc meaning.

Reply:
Right. This has disappeared in the new version :)